RELEASE.md

```diff
@@ -1,5 +1,17 @@
 # MPI Operator Releases
 
+## Release v0.5.0
```

Comment:
Could we mention that we have started to support K8s v1.28 and v1.29?

Reply:
uhm... it should have run before. But I can mention it in the cleanups.

Are we missing any PR before I rebase?

Reply:
> uhm... it should have run before. But I can mention it in the cleanups.

That makes sense.

> Are we missing any PR before I rebase?

The last piece is https://github.com/kubeflow/mpi-operator/pull/634.